### PR TITLE
ADD: Timestamps to DB Objects

### LINKED ISSUE #97: Add "Datum" to "FileRef" Database Model

### PR DESCRIPTION
**Pull request by iTitus**
Closes #97

### CODE REVIEW

File: src/main/java/mops/persistence/permission/DirectoryPermissions.java [hunk 1]
```diff
@@ -48,8 +66,25 @@ public DirectoryPermissions(@NonNull Set<DirectoryPermissionEntry> permissions)
...
+    }
+
+    /**
+     * Get the last modified time.
+     *
+     * @return last modified time
+     */
+    public Instant getLastModifiedTime() {
+        return lastModifiedTime == null ? Instant.EPOCH : lastModifiedTime.toInstant();
     }
```
Comment (Segelzwerg) [inline src/main/java/mops/persistence/permission/DirectoryPermissions.java]: Do you test this?

File: src/main/java/mops/presentation/DirectoryController.java [hunk 2]
```diff
@@ -118,7 +117,7 @@ public String deleteFolder(KeycloakAuthenticationToken token,
     public String searchFolder(KeycloakAuthenticationToken token,
                                Model model,
                                @PathVariable("dirId") long dirId,
-                               @ModelAttribute("searchQuery") FileQuery query) {
+                               @RequestAttribute("searchQuery") FileQuery query) {
```
Comment (Segelzwerg) [inline src/main/java/mops/presentation/DirectoryController.java]: Can you explain why `@RequestAttribute` I thought this isn't working. Not critic just curious.

File: src/main/java/mops/persistence/directory/Directory.java [hunk 3]
```diff
@@ -47,10 +62,25 @@
      * @param groupOwner    Id of the owing group
```
Comment (KaFe23) [inline src/main/java/mops/persistence/directory/Directory.java]:
```suggestion
     * @param groupOwner    Id of the owning group
```

File: src/main/java/mops/persistence/permission/DirectoryPermissions.java (hunk 1 above)
Comment (iTitus) [reply src/main/java/mops/persistence/permission/DirectoryPermissions.java]: just a simple getter, gets implicitly tested by the equals

**Pull request merged**